adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module Middlewares
+        class TracerMiddleware < ::Faraday::Middleware
+          def call(env)
+            tracer.in_span(env.url.to_s,
+                          attributes: { 'component' => 'http',
+                                        'http.method' => env.method,
+                                        'http.url' => env.url.to_s },
+                          kind: :client) do |span|
+              propagate_context(span, env)
+
+              app.call(env).on_complete { |resp| trace_response(span, resp) }
+            end
+          end
+
+          private
+
+          attr_reader :app
+
+          def propagate_context(span, env)
+            formatter.extract(env)
+            formatter.inject(span.context, env.request_headers)
```

Comment:
We only want the `inject` here, not the `extract`. For outbound requests, we *inject* the current span context into the request headers. For an inbound request (e.g. in the Sinatra adapter), we *extract* the remote span context from the request headers and start a new span with the extracted context as the parent context (`with_parent_context`).

Reply:
Thanks for that guidance!  I've made the relevant changes for this PR, and will need to check #149 as well :heavy_check_mark: